### DISCUSSION TITLE
Stripe incomplete invoices

### DISCUSSION THREAD

**Discussion by Rainneee** [pay-rails/pay#957] [Q&A]
Hi @excid3, I noticed in my Jumpstart Pro app that a view to payment creates a Stripe invoice/subscription to even render the page. When I had done Stripe integrations myself, I did not have that issue, so I went to investigate. I saw in your code you have this comment: # Stripe subscriptions are handled entirely client-side # We need to create a subscription to render the PaymentElement def new. So if people visit the checkout page multiple times without paying, how should we handle these empty invoices? Just check Stripe on our own and delete them? This seems like a suboptimal way to do this.

Is there anyone with the same issue? How do you go about this?

**Comment by excid3** [comment]
Stripe will automatically delete incomplete subscriptions after a period of time so there is nothing you need to do.

We don't use the PaymentElement in Jumpstart Pro anymore btw.

**Reply by Rainneee** [reply]
Thanks for the response @excid3, but we havent synced the jumpstart code base in a while. if we sync, will we need to change around our payment code or will it automatically just work?

**Reply by excid3** [reply]
Depends on what you've changed?
